include/zephyr/net/icmp.h

```diff
@@ -46,6 +46,44 @@ typedef int (*net_icmp_handler_t)(struct net_icmp_ctx *ctx,
 				  struct net_icmp_hdr *icmp_hdr,
 				  void *user_data);
 
+/**
+ * @typedef net_icmp_offload_ping_handler_t
+ * @brief Handler function that is called when an Echo-Request is sent
+ *        to offloaded device. This handler is typically setup by the
+ *        device driver so that it can catch the ping request and send
+ *        it to the offloaded device.
+ *
+ * @param ctx ICMP context used in this request.
+ * @param iface Network interface, can be set to NULL in which case the
+ *        interface is selected according to destination address.
+ * @param dst IP address of the target host.
+ * @param identifier An identifier to aid in matching Echo Replies
+ *        to this Echo Request. May be zero.
+ * @param sequence A sequence number to aid in matching Echo Replies
+ *        to this Echo Request. May be zero.
+ * @param tc_tos Can be either IPv4 Type-of-service field value,
+ *        or IPv6 Traffic Class field value. Represents combined DSCP and
+ *        ECN values.
+ * @param priority Network packet priority.
+ * @param data Arbitrary payload data that will be included in the
+ *        Echo Reply verbatim. May be NULL.
+ * @param data_size Size of the Payload Data in bytes. May be zero. In case data
+ *        pointer is NULL, the function will generate the payload up to the requested
+ *        size.
+ * @param user_data User supplied opaque data passed to the handler. May be NULL.
+ *
+ */
+typedef int (*net_icmp_offload_ping_handler_t)(struct net_icmp_ctx *ctx,
+					       struct net_if *iface,
+					       struct sockaddr *dst,
+					       uint16_t identifier,
+					       uint16_t sequence,
+					       uint8_t tc_tos,
+					       int priority,
+					       const void *data,
+					       size_t data_size,
```

Comment:
Perhaps it'd make sense to define `struct net_icmp_ping_params` and use it as a function argument instead of repeating the lengthy parameter list?

Reply:
Yes, makes sense. I will fix this.